packages/storage-plus/src/de.rs

```diff
@@ -0,0 +1,157 @@
+            }
+        })*
+    }
+}
+
+string_de!(for String, &String, str, &str, Addr, &Addr);
+
+macro_rules! integer_de {
+    (for $($t:ty),+) => {
+        $(impl Deserializable for IntKey<$t> {
+            type Output = $t;
+
+            fn from_slice(value: &[u8]) -> StdResult<Self::Output> {
+                Ok(<$t>::from_be_bytes(value.try_into()
+                    // FIXME: Add and use StdError try-from error From helper
+                    .map_err(|err: TryFromSliceError| StdError::generic_err(err.to_string()))?))
+            }
+
+            fn from_vec(value: &Vec<u8>) -> StdResult<Self::Output> {
+                Ok(<$t>::from_be_bytes(value.to_vec().try_into().unwrap()))
+            }
+        })*
+    }
+}
+
+integer_de!(for i8, u8, i16, u16, i32, u32, i64, u64, i128, u128);
+
+impl<T: Deserializable, U: Deserializable> Deserializable for (T, U) {
+    type Output = (T::Output, U::Output);
+
+    fn from_slice(value: &[u8]) -> StdResult<Self::Output> {
+        let t_len = u16::from_be_bytes(
+            value[..2]
+                .try_into()
+                // FIXME: Add and use StdError try-from error From helper
+                .map_err(|err: TryFromSliceError| StdError::generic_err(err.to_string()))?,
+        ) as usize;
+        let t = T::from_slice(&value[2..2 + t_len])?;
+        let u = U::from_slice(&value[2 + t_len..])?;
+
+        Ok((t, u))
+    }
+
+    fn from_vec(value: &Vec<u8>) -> StdResult<Self::Output> {
+        let t_len = u16::from_be_bytes(
+            value[..2]
+                .try_into()
+                // FIXME: Add and use StdError try-from error From helper
+                .map_err(|err: TryFromSliceError| StdError::generic_err(err.to_string()))?,
+        ) as usize;
+        let t = T::from_slice(&value[2..2 + t_len])?;
+        let u = U::from_slice(&value[2 + t_len..])?;
+
+        Ok((t, u))
+    }
+}
+
+impl<T: Deserializable, U: Deserializable, V: Deserializable> Deserializable for (T, U, V) {
+    type Output = (T::Output, U::Output, V::Output);
+
+    fn from_slice(value: &[u8]) -> StdResult<Self::Output> {
+        let t_len = u16::from_be_bytes(
+            value[..2]
+                .try_into()
+                // FIXME: Add and use StdError try-from error From helper
+                .map_err(|err: TryFromSliceError| StdError::generic_err(err.to_string()))?,
+        ) as usize;
+        let t = T::from_slice(&value[2..2 + t_len])?;
+        let u_len = u16::from_be_bytes(
+            value[2 + t_len..4 + t_len]
+                .try_into()
+                // FIXME: Add and use StdError try-from error From helper
+                .map_err(|err: TryFromSliceError| StdError::generic_err(err.to_string()))?,
+        ) as usize;
+        let u = U::from_slice(&value[4 + t_len..4 + t_len + u_len])?;
+        let v = V::from_slice(&value[4 + t_len + u_len..])?;
+
+        Ok((t, u, v))
+    }
+
+    fn from_vec(value: &Vec<u8>) -> StdResult<Self::Output> {
```

Comment:
If these are the same implementation, and we know we can Deref Vec to &[u8], can't you just implement as

```rust
fn from_vec(value: &Vec<u8>) -> StdResult<Self::Output> {
  Self::from_slice(&value)
}
```

Reply:
Good. I think that's what I was trying to achieve. But then this is probably no better than the original `from_slice` impl.